### PR TITLE
Connecting to a newer node triggers autoupgrade check (fixes #1177)

### LINKED ISSUE #1177: New version notification could alert other known nodes
Today one of my nodes noticed v0.10.16 was available. Maybe it could poke the other nodes to tell them to check for an update. Or maybe they could notice when a newer node contacts them.

### PR DESCRIPTION
**Pull request by AudriusButkevicius**
DDoS github!

By the way, I hope you don't mind my commits traveling back in time.
Aiming for a stupid 1 commit a day, and need to cover up for dem
holidays.

### CODE REVIEW

File: internal/model/model.go [hunk 1]
```diff
@@ -575,6 +579,12 @@ func (m *Model) ClusterConfig(deviceID protocol.DeviceID, cm protocol.ClusterCon
 	}
 	m.pmut.Unlock()
 
+	if m.upgradeTimer != nil && upgrade.CompareVersions(cm.ClientVersion, m.clientVersion) == upgrade.Newer {
```
Comment (calmh) [inline internal/model/model.go]: We should probably check the client name here as well, so we only do this for higher versions of syncthing, not for "Pulse v42" or whatever.

**Comment by calmh** [comment]
Looks reasonable to me, apart from not being really sure that it's worth it and the poor Model becoming even more of a swamp of dependencies... :/ :)

**Comment by calmh** [comment]
Hmm, how about using the event system for this? A routine (maybe the auto upgrade loop itself) that subscribes to the `DeviceConnected` event and does a check when a connect comes in from a higher version?

**Comment by calmh** [comment]
(*Maybe* we want some rate limiting here, as Github starts returning 403 to the upgrade checks when there are more than 60 in the last 60 minutes. But it's not exactly essential.)

**Comment by AudriusButkevicius** [comment]
Re events, good idea.
Re rate limiting it's not necessary as you have 60 requests per IP per hour, and this PR increments possible lookus by one only.

**Comment by AudriusButkevicius** [comment]
As per your suggestions now.

File: internal/model/model.go [hunk 2]
```diff
@@ -577,8 +577,23 @@ func (m *Model) ClusterConfig(deviceID protocol.DeviceID, cm protocol.ClusterCon
...
+
+	event := map[string]string{
+		"id":            deviceID.String(),
+		"clientName":    cm.ClientName,
+		"clientVersion": cm.ClientVersion,
+	}
+
+	conn, ok := m.rawConn[deviceID]
+	if ok {
+		if conn, ok := conn.(*tls.Conn); ok {
```
Comment (calmh) [inline internal/model/model.go]: shorter:

```
if conn,ok := m.rawConn[deviceID].(net.Conn); ok {
...
```

File: internal/events/events.go [hunk 3]
```diff
@@ -98,10 +98,10 @@ type Event struct {
 }
 
 type Subscription struct {
-	mask   EventType
-	id     int
-	events chan Event
-	mutex  sync.Mutex
+	mask  EventType
+	id    int
+	C     chan Event
```
Comment (calmh) [inline internal/events/events.go]: Hmm. In this way expose a writable channel. Perhaps keep `events` as it is, and add a 

```
func (s* Subscription) C() <-chan Event {
return s.events
}
```

?

**Pull request merged**